samples/ClassroomLabs/Scripts/GenericPreparation/Prepare-MicrosoftStoreApplications.ps1

```diff
@@ -0,0 +1,133 @@
+    Write-Verbose "Running with Administrator privileges (t/f): $isAdministrator"
+    return $isAdministrator
+}
+
+
+<#
+.SYNOPSIS
+Removes selected Microsoft Store applications.  Uninstallable applications are not shown.
+#>
+function Remove-MicrosoftStoreApps{
+
+    $options =  $(New-Object -Type 'System.Management.Automation.Host.ChoiceDescription' -ArgumentList @('&Select specific applications', 'Select specific applications to delete from a grid.')), `
+                $(New-Object -Type 'System.Management.Automation.Host.ChoiceDescription' -ArgumentList @('&None','Do not remove any Microsoft Store applications.'))
+                
+    $response = $Host.UI.PromptForChoice("Remove Microsoft Store applications", 'Which Microsoft Store applications should be removed?', $options, 0)
+
+    if ($response -eq 0){
+        
+        $microsoftStoreAppsToRemove= Get-AppxPackage | Where {$_.NonRemovable -eq $false} | Sort-Object -Property Name  | Out-Gridview -Title "Select which Microsoft Store applications to delete" -PassThru
+        $appsThatAreDependencies = @(Get-AppxPackage | Select -expand Dependencies | select -expand PackageFullName)  
+        #Remove Microsoft Store application, as long as it is not a dependency of another application.         
+        $microsoftStoreAppsToRemove | `
+            %{
+                if ($appsThatAreDependencies -contains $_.PackageFullName){
+                    $currentPackageFullName = $_.PackageFullName
+                    $appsWithDependencyOnCurrentPackageList =  @(Get-AppxPackage | where {$($_.Dependencies | select -expand PackageFullName) -eq $currentPackageFullName} | select -expand Name)
+                    Write-Host "Can't remove '$($_.Name)' as it is a dependency for $($appsWithDependencyOnCurrentPackageList -join ', ')."
+                }else{
+                    %{Write-Host "Removing '$($_.Name)'."; $_ | Remove-AppxPackage }
+                }
+            }
+    }else{
+        Write-Host "No Microsoft Store applications removed."
+    }
+}
+
+<#
+.SYNOPSIS
+Updates Microsoft Store applications.  
+#>
+function Update-MicrosoftStoreApps{
+    Write-Host "Updating Microsoft Store applications."
+    (Get-WmiObject -Namespace "root\cimv2\mdm\dmmap" -Class "MDM_EnterpriseModernAppManagement_AppManagement01").UpdateScanMethod() | Out-Null
+}
+
+###################################################################################################
+#
+# Main execution block.
+#
+
+try {
+    Write-Host "Verifying running as administrator."
+    if (-not (Get-RunningAsAdministrator)) { 
+        Write-Error "Please re-run this script as Administrator." 
+    }
+
+    Write-Host "Removing Microsoft Store apps that are not needed for class.  Any remaining apps will be updated."
+    Remove-MicrosoftStoreApps
+
+    Write-Host "Updating Microsoft Store apps."
+    Update-MicrosoftStoreApps
+  
+    Write-Host "Script completed." -ForegroundColor Green

```

Comment:
Consider writing out success as well and format the message similar to that in your error handling logic. For example,
``` PS1
Write-Host "The script completed successfully." -ForegroundColor Green
```